src/main/java/com/avispl/symphony/dal/infrastructure/management/nec/navisetadministrator2se/NaViSetAdministrator2SECommunicator.java

```diff
@@ -0,0 +1,1251 @@
+	 * this variable stores a timestamp which validates it, so when the devices' statistics is done collecting, variable
+	 * is set to currentTime + 30s, at the same time, calling {@link #retrieveMultipleStatistics()} and updating the
+	 */
+	private long nextDevicesCollectionIterationTimestamp;
+
+	/**
+	 * This parameter holds timestamp of when we need to stop performing API calls
+	 * It used when device stop retrieving statistic. Updated each time of called #retrieveMultipleStatistics
+	 */
+	private volatile long validRetrieveStatisticsTimestamp;
+
+	/**
+	 * Aggregator inactivity timeout. If the {@link NaViSetAdministrator2SECommunicator#retrieveMultipleStatistics()}  method is not
+	 * called during this period of time - device is considered to be paused, thus the Cloud API
+	 * is not supposed to be called
+	 */
+	private static final long retrieveStatisticsTimeOut = 3 * 60 * 1000;
+
+	/**
+	 * Executor that runs all the async operations, that is posting and
+	 */
+	private ExecutorService executorService;
+
+	/**
+	 * A private field that represents an instance of the NavisetDataLoader class, which is responsible for loading device data for Naviset
+	 */
+	private NavisetDataLoader deviceDataLoader;
+
+	/**
+	 * A JSON node containing the response from an aggregator.
+	 */
+	private JsonNode aggregatorResponse;
+
+	/**
+	 * API Token
+	 */
+	private String token;
+
+	/**
+	 * Configurable property for historical properties, comma separated values kept as set locally
+	 */
+	private Set<String> historicalProperties = new HashSet<>();
+
+	/**
+	 * Retrieves {@link #historicalProperties}
+	 *
+	 * @return value of {@link #historicalProperties}
+	 */
+	public String getHistoricalProperties() {
+		return String.join(",", this.historicalProperties);
+	}
+
+	/**
+	 * Sets {@link #historicalProperties} value
+	 *
+	 * @param historicalProperties new value of {@link #historicalProperties}
+	 */
+	public void setHistoricalProperties(String historicalProperties) {
+		this.historicalProperties.clear();
+		Arrays.asList(historicalProperties.split(",")).forEach(propertyName -> {
+			this.historicalProperties.add(propertyName.trim());
+		});
+	}
+
+	/**
+	 * Update the status of the device.
+	 * The device is considered as paused if did not receive any retrieveMultipleStatistics()
+	 * calls during {@link NaViSetAdministrator2SECommunicator}
+	 */
+	private synchronized void updateAggregatorStatus() {
+		devicePaused = validRetrieveStatisticsTimestamp < System.currentTimeMillis();
+	}
+
+	/**
+	 * Uptime time stamp to valid one
+	 */
+	private synchronized void updateValidRetrieveStatisticsTimestamp() {
+		validRetrieveStatisticsTimestamp = System.currentTimeMillis() + retrieveStatisticsTimeOut;
+		updateAggregatorStatus();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 * <p>
+	 *
+	 * Check for available devices before retrieving the value
+	 * ping latency information to Symphony
+	 */
+	@Override
+	public int ping() throws Exception {
```

Comment:
alternative TCP ping approach has somewhat backfired on prod, so some adapters that support it, break basic monitoring (because no interfaces were setup, which defaults port number to 0). 
If this is really what we need for these devices, it'd be great to add some kind of trigger, that would allow switching between default ICMP ping and TCP ping